src/librustc/diagnostics.rs

```diff
@@ -112,6 +112,50 @@ reference when using guards or refactor the entire expression, perhaps by
 putting the condition inside the body of the arm.
 "##,
 
+E0009: r##"
+In a pattern, all values that don't implement the `Copy` trait have to be bound
+the same way. The goal here is to avoid binding simultaneous by-move and by-ref
+at the same time.
```

Comment:
`The goal here is to avoid binding simultaneously by move and by reference.`

I'd also like a note about this restriction possibly being lifted:

`This limitation may be removed in a future version of Rust.`